spec/services/facility_stats_service_spec.rb

```diff
@@ -39,81 +22,136 @@ def facilities_data(facilities)
+                                                period: period, rate_numerator: :controlled_patients)
+      period_keys = stats_by_size[:small][:periods].values.map(&:keys).flatten.uniq
       controlled_patient_keys = ["controlled_patients", "controlled_patients_rate"]
       expect(period_keys & controlled_patient_keys).to match_array(controlled_patient_keys)
     end
 
     it "processes data for uncontrolled_patients" do
       stats_by_size = FacilityStatsService.call(facilities: facilities_data([small_facility]),
-                                                period: period, rate_numerator: "uncontrolled_patients")
-      period_keys = stats_by_size["small"].values.map(&:keys).flatten.uniq
+                                                period: period, rate_numerator: :uncontrolled_patients)
+      period_keys = stats_by_size[:small][:periods].values.map(&:keys).flatten.uniq
       uncontrolled_patient_keys = ["uncontrolled_patients", "uncontrolled_patients_rate"]
       expect(period_keys & uncontrolled_patient_keys).to match_array(uncontrolled_patient_keys)
     end
 
     it "processes data for missed_visits" do
       stats_by_size = FacilityStatsService.call(facilities: facilities_data([small_facility]),
-                                                period: period, rate_numerator: "missed_visits")
-      period_keys = stats_by_size["small"].values.map(&:keys).flatten.uniq
+                                                period: period, rate_numerator: :missed_visits)
+      period_keys = stats_by_size[:small][:periods].values.map(&:keys).flatten.uniq
       missed_visits_keys = ["missed_visits", "missed_visits_rate"]
       expect(period_keys & missed_visits_keys).to match_array(missed_visits_keys)
     end
 
     it "handles invalid rate_numerator by setting values to zero" do
       stats_by_size = FacilityStatsService.call(facilities: facilities_data([small_facility]),
-                                                period: period, rate_numerator: "womp")
-      stat_keys = stats_by_size["small"].values.first.keys
-      stat_values = stats_by_size["small"].values.first.values
+                                                period: period, rate_numerator: :womp)
+      stat_keys = stats_by_size[:small][:periods].values.first.keys
+      stat_values = stats_by_size[:small][:periods].values.first.values
       expected_keys = ["womp", "adjusted_registrations", "cumulative_registrations", "womp_rate"]
       expect(stat_keys).to match_array(expected_keys)
       expect(stat_values).to match_array([0, 0, 0, 0])
     end
+
+    it "sets total_registered_hypertensive_patients for each size" do
+      small_facility
+      month = december - 4.months
+      medium_facility1 = create(:facility, name: "medium_1", facility_size: "medium")
+      medium_facility2 = create(:facility, name: "medium_2", facility_size: "medium")
+      create_list(:patient, 2, :hypertension, full_name: "medium_uncontrolled",
```

Comment:
The biggest challenge I faced in this ticket was figuring out how to get patient data that showed up in the `facility.registered_hypertension_patients`. For reasons not clear to me yet, setting up the test data like I did above (make facility, make patients for facility, add blood pressures for patients) results in `facility.registered_hypertension_patients` being empty. Conversely, setting up patients with traits as I've done here results in the patient data not appearing in the RegionService controlled/uncontrolled results (I believe this was true even if I also created blood pressure readings). The problem may be my lack of understanding of how that data gets into the controlled/uncontrolled results, but I suspect there's something missing with the factories. Ideally, I would prefer to use these factories with traits, because they set up a medical history, which results in less tedious data setup (no need to create BP readings a month after setting up the patient).